logstash-core/src/main/java/org/logstash/settings/SettingNumeric.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.logstash.settings;
+
+public class SettingNumeric extends Coercible<Number> {
+
+    public SettingNumeric(String name, Number defaultValue) {
+        super(name, defaultValue, true, noValidator());
+    }
+
+    // constructor used only in tests, but needs to be public to be used in Ruby spec
+    public SettingNumeric(String name, Number defaultValue, boolean strict) {
+        super(name, defaultValue, strict, noValidator());
+    }
+
+    @Override
+    public Number coerce(Object obj) {
```

Comment:
It seems the Ruby's version is handling `nil` obj values, don't we need to do the same here for `null`?

Reply:
From `irb` REPL on previous code I got:
```
> s.coerce(:nil)
/Users/andrea/workspace/logstash_andsel/logstash-core/lib/logstash/settings.rb:430:in `coerce': Failed to coerce value to Numeric. Received nil (Symbol) (ArgumentError)
```

so you are suggesting to add something
```java
@Override
    public Number coerce(Object obj) {
      if (obj == null) {
        throw new IllegalArgumentException("Failed to coerce value to SettingNumeric. Received null");
      }
}
```

Reply:
Fixed by e82f5903793d31811dabb7d7a31b1b2979b33578